Src/Plugins/Security/SEC101_043.NuGetCredentialsValidator.cs

```diff
@@ -0,0 +1,154 @@
+
+        static NuGetCredentialsValidator()
+        {
+            Instance = new NuGetCredentialsValidator();
+        }
+
+        public static ValidationState IsValidStatic(ref string matchedPattern,
+                                                    ref Dictionary<string, string> groups,
+                                                    ref string failureLevel,
+                                                    ref string message,
+                                                    out Fingerprint fingerprint)
+        {
+            return ValidatorBase.IsValidStatic(Instance,
+                                               ref matchedPattern,
+                                               ref groups,
+                                               ref failureLevel,
+                                               ref message,
+                                               out fingerprint);
+        }
+
+        public static ValidationState IsValidDynamic(ref Fingerprint fingerprint, ref string message, ref Dictionary<string, string> options)
+        {
+            return IsValidDynamic(Instance,
+                                  ref fingerprint,
+                                  ref message,
+                                  ref options);
+        }
+
+        protected override ValidationState IsValidStaticHelper(ref string matchedPattern,
+                                                               ref Dictionary<string, string> groups,
+                                                               ref string failureLevel,
+                                                               ref string message,
+                                                               out Fingerprint fingerprint)
+        {
+            fingerprint = default;
+
+            if (!groups.TryGetNonEmptyValue("secret", out string secret) ||
+                !groups.TryGetNonEmptyValue("id", out string id) ||
+                !groups.TryGetNonEmptyValue("host", out string host))
+            {
+                return ValidationState.NoMatch;
+            }
+
+            if (StaticStringHelper.LikelyPowershellVariable(secret))
+            {
+                return ValidationState.NoMatch;
+            }
+
+            fingerprint = new Fingerprint
+            {
+                Host = host, // technically, this is XML which could contain multiple hosts.
+                Id = id,
+                Secret = secret,
+                Platform = nameof(AssetPlatform.NuGet),
+            };
+
+            return ValidationState.Unknown;
+        }
+
+        protected override ValidationState IsValidDynamicHelper(ref Fingerprint fingerprint, ref string message, ref Dictionary<string, string> options)
+        {
+            string hostXmlAsString = fingerprint.Host;
+            string username = fingerprint.Id;
+            string password = fingerprint.Secret;
+
+            XmlDocument hostXml = new XmlDocument();
```

Comment:
>XmlDocument [](start = 12, length = 11)

this needs to be 'var'. do me a favor: consult with Eddy about getting your style settings corrected so that VS will tell you about the issue. this is one of those style nits that is very fatiguing to receive and to give in a code review context, in my opinion.

Reply:
I'm meeting with him in a minute about it.  Just to clarify, it's not that it doesn't notify me.  It notifies me *of the exact opposite*, telling me to use an explicit type, with a red mark.  I'm not sure if it blocks building or anything because, as you can tell, I've been doing whatever auto-correct says.  We'll try to figure it out in a bit.

---
In reply to: [611238763](https://github.com/microsoft/sarif-pattern-matcher/pull/372#discussion_r611238763) [](ancestors = 611238763)